ethereum/forge-test/GasBenchmark.t.sol

```diff
@@ -0,0 +1,139 @@
+            sequence,
+            NO_SIGNERS
+        );
+
+        ++sequence;
+        
+        updateData = new bytes[](1);
+        updateData[0] = vaa;
+
+        updateFee = pyth.getUpdateFee(updateData);
+    }
+
+    function testBenchmarkUpdatePriceFeedsFresh() public {
+        for (uint i = 0; i < BENCHMARK_ITERATIONS; ++i) {
+            advancePrices();
+
+            (bytes[] memory updateData, uint updateFee) = generateUpdateDataAndFee();
+            pyth.updatePriceFeeds{value: updateFee}(updateData);
+        }
+    }
+
+    function testBenchmarkUpdatePriceFeedsNotFresh() public {
+        for (uint i = 0; i < BENCHMARK_ITERATIONS; ++i) {
+            (bytes[] memory updateData, uint updateFee) = generateUpdateDataAndFee();
+            pyth.updatePriceFeeds{value: updateFee}(updateData);
+        }
+    }
+
+    function testBenchmarkUpdatePriceFeedsIfNecessaryFresh() public {
+        for (uint i = 0; i < BENCHMARK_ITERATIONS; ++i) {
+            advancePrices();
+
+            uint64[] memory publishTimes = new uint64[](NO_PRICES);
+            
+            for (uint j = 0; j < NO_PRICES; ++j) {
+                publishTimes[j] = uint64(prices[j].publishTime);
+            }
+
+            (bytes[] memory updateData, uint updateFee) = generateUpdateDataAndFee();
+
+            // Since the prices have advanced, the publishTimes are newer than one in
+            // the contract and hence, the call should succeed.
+            pyth.updatePriceFeedsIfNecessary{value: updateFee}(updateData, priceIds, publishTimes);
+        }
+    }
+
+    function testBenchmarkUpdatePriceFeedsIfNecessaryNotFresh() public {
+        for (uint i = 0; i < BENCHMARK_ITERATIONS; ++i) {
+            uint64[] memory publishTimes = new uint64[](NO_PRICES);
+            
+            for (uint j = 0; j < NO_PRICES; ++j) {
+                publishTimes[j] = uint64(prices[j].publishTime);
+            }
+
+            (bytes[] memory updateData, uint updateFee) = generateUpdateDataAndFee();
+
+            // Since the price is not advanced, the publishTimes are the same as the
+            // ones in the contract except the first update.
+            if (i > 0) {
+                vm.expectRevert(bytes("no prices in the submitted batch have fresh prices, so this update will have no effect"));
+            }
+    
+            pyth.updatePriceFeedsIfNecessary{value: updateFee}(updateData, priceIds, publishTimes);
+        }
+    }
+
+    function testBenchmarkGetPrice() public {
+        (bytes[] memory updateData, uint updateFee) = generateUpdateDataAndFee();
+        pyth.updatePriceFeeds{value: updateFee}(updateData);
+
+        // Set the block timestamp to the publish time, so getPrice work as expected.
+        vm.warp(prices[0].publishTime);
+
+        for (uint i = 0; i < BENCHMARK_ITERATIONS; ++i) {
+            PythStructs.Price memory price = pyth.getPrice(priceIds[i % NO_PRICES]);
+            emit log_named_int("price", price.price);
```

Comment:
Curious why we emit an event here? Will this make the benchmark less accurate?

Reply:
I wanted to make sure that this part won't be removed by the optimizer.

Reply:
You could increment an accumulator instead, might be cheaper.

Reply:
Good point. will do!

Reply:
I checked and it was not getting optimized (probably cross contract calls won't get optimized). So i removed it entirely.